File: WatchFramework/GameLogic.m
//
//  GameLogic.m
//  Watch Match
//
//  Created by Watch Set Team on 11/23/14.
//  Copyright (c) 2014 Watch Set Team. All rights reserved.
//

#import "GameLogic.h"

#import "GameTurn.h"

@implementation GameLogic

- (instancetype)init {
    self = [super init];
    if (self != nil) {
        NSBundle *mainBundle = [NSBundle mainBundle];
        NSURL *URL = [mainBundle URLForResource:@"Game" withExtension:@"plist"];
        
        NSDictionary *dictionary = [NSDictionary dictionaryWithContentsOfURL:URL];
        
        _startTime = dictionary[@"time"];
        
        NSArray *turnDictionariesArray = dictionary[@"turns"];
        
        NSMutableArray *turns = [[NSMutableArray alloc] init];
        for (NSDictionary *turnDictionary in turnDictionariesArray) {
            GameTurn *turn = [[GameTurn alloc] initWithDictionary:turnDictionary];
            
            [turns addObject:turn];
        }
        
        _turns = turns;
    }
    return self;
}

@end

File: WatchFramework/GameLogic.swift
//
//  GameLogic.swift
//  Watch Match
//
//  Created by Andrew Pontious on 11/26/14.
//  Copyright (c) 2014 TeamHand. All rights reserved.
//

import Foundation

public class GameLogic {
    
    public let turns: [GameTurn]
    public let startTime: NSDate

    public init() {
        // Assumes Game.plist file is valid
        
        let mainBundle = NSBundle.mainBundle()
        let URL = mainBundle.URLForResource("Game", withExtension:"plist")!
        
        let dictionary = NSDictionary(contentsOfURL:URL) as NSDictionary!
        
        var turns = [GameTurn]()
        
        self.startTime = dictionary["time"] as NSDate
        
        let turnDictionaries = dictionary["turns"] as [NSDictionary]
        
        for (index, turnDictionary) in enumerate(turnDictionaries) {
            let turn = GameTurn(dictionary: turnDictionary);
            
            turns.append(turn)
        }
        
        self.turns = turns
    }
}